src/series/algorithm/waterfall.js

```diff
@@ -0,0 +1,107 @@
+import d3 from 'd3';
+import {noop} from '../../util/fn';
+
+export default function() {
+
+    var xValueKey = '',
+        yValue = function(d) { return d.y; },
+        startsWithTotal = false,
+        totals = noop;
+
+    var waterfall = function(data) {
+        var length = data.length,
+            i = 0,
+            previousEnd = yValue(data[0]),
+            start,
+            end,
+            total,
+            result = [],
+            finalIndex = length - 1,
+            customTotals = false;
+
+        if (startsWithTotal) {
+            // First value is a total
+            result.push({
+                domain: data[0][xValueKey],
+                start: 0,
+                end: previousEnd,
+                direction: 0
```

Comment:
The 0 isn't so much 'unchanged' (it can be), but demarks a totals column which has its css default to a standard bar colour. Perhaps that logic can be described by 'unchanged' though.